compiler/z/codegen/FPTreeEvaluator.cpp

```diff
@@ -1268,24 +1241,30 @@ OMR::Z::TreeEvaluator::fbits2iEvaluator(TR::Node * node, TR::CodeGenerator * cg)
           generateLoadLiteralPoolAddress(cg, node, litBase);
           }
 
-       TR::MemoryReference * NaN              = generateS390MemoryReference((int32_t)0x7fc00000, TR::Int32, cg, litBase);
+       TR::MemoryReference * NaN              = nodeType == TR::Int64 ? generateS390MemoryReference((int64_t)CONSTANT64(0x7ff8000000000000), TR::Int64, cg, litBase) : 
+                                                                           generateS390MemoryReference((int32_t)0x7fc00000, TR::Int32, cg, litBase);
```

Comment:
While we're in the area let's introduce some formatting here:

```
TR::MemoryReference* nanMemRef = nodeType == TR::Int64 ? 
          generateS390MemoryReference(static_cast<int64_t>(0x7ff8000000000000), nodeType, cg, litBase) : 
          generateS390MemoryReference(static_cast<int32_t>(0x7fc00000),         nodeType, cg, litBase);
```

Reply:
3ba28f5190d6507418b1678a02e6ee7a359bc67e